src/common/data.py

```diff
@@ -51,3 +51,105 @@ def generate(self):
+                 n_targets: int,
+                 bias: float,
+                 noise: float,
+                 seed: int):
+        """Initializes the generator"""
+        self.n_label_classes = n_label_classes
+        self.batch_size = batch_size
+        self.n_features = n_features
+        self.n_informative = min(n_features, n_informative)
+        self.n_targets = n_targets
+        self.bias = bias
+        self.noise = noise
+        self.generator = np.random.RandomState(seed)
+
+        # Generate a ground truth model with only n_informative features being non-zeros
+        self.ground_truth = np.zeros((self.n_features, self.n_targets))
+        self.ground_truth[:n_informative, :] = 100 * self.generator.rand(self.n_informative, self.n_targets)
+
+    def generate(self):
+        """Generate one batch of data.
+
+        Returns:
+            X (numpy.ndarray)
+            y (numpy.ndarray)
+        """
+        # Randomly generate a well conditioned input set
+        X = self.generator.randn(self.batch_size, self.n_features)
+        y = np.dot(X, self.ground_truth) + self.bias
+
+        # Add noise
+        if self.noise > 0.0:
+            y += self.generator.normal(scale=self.noise, size=y.shape)
+
+        # create n_label_classes ranking labels
+        y = ((y - min(y))/(max(y)-min(y))*self.n_label_classes).astype(int)
+
+        y = np.squeeze(y)
+
+        return X, y
+
+
+class RankingDataGenerator():
+    """Generator for regression data"""
+    def __init__(self,
+                 docs_per_query: int,
+                 n_label_classes: int,
+                 batch_size: int,
+                 n_features: int,
+                 n_informative: int,
+                 n_targets: int,
+                 bias: float,
+                 noise: float,
+                 seed: int):
+        """Initializes the generator"""
+        self.docs_per_query = docs_per_query
+        self.n_label_classes = n_label_classes
+        self.batch_size = batch_size
+        self.n_features = n_features
+        self.n_informative = min(n_features, n_informative)
+        self.n_targets = n_targets
+        self.bias = bias
+        self.noise = noise
+        self.generator = np.random.RandomState(seed)
+
+        # Generate a ground truth model with only n_informative features being non-zeros
+        self.ground_truth = np.zeros((self.n_features, self.n_targets))
+        self.ground_truth[:n_informative, :] = 100 * self.generator.rand(self.n_informative, self.n_targets)
+
+    def generate(self):
+        """Generate one batch of data.
+
+        Returns:
+            X (numpy.ndarray)
+            y (numpy.ndarray)
+        """
+        # Randomly generate a well conditioned input set
+        X = self.generator.randn(self.batch_size, self.n_features)
+        y = np.dot(X, self.ground_truth) + self.bias
+
+        # add query column
+        query_col = [[i // self.docs_per_query] for i in range(self.batch_size)]
```

Comment:
so in this situation, each batch contains docs_per_query identical query ids. but that repeats for each batch, so ultimately the dataset contains a lot of the same query ids.

since we're not using distributed for now, maybe a solution could be to introduce a batch count and use it to generate differnt query ids for each batch?

Reply:
fixed